packages/office-ui-fabric-react/src/components/ScreenReaderAlert/ScreenReaderAlert.tsx

```diff
@@ -0,0 +1,163 @@
+ * 2. JAWS 18.
+ * 3. NVDA 2016.4.
+ * 4. ChromeVOX v53. (Not support ReadingMode.ReadAfterOtherContent option for ChromeVOX, it will read immediately)
+ *
+ * @example
+ * <ScreenReaderAlert>
+ *   { yourFirstAlertText }
+ *   { yourSecondAlertText }
+ * </ScreenReaderAlert>
+ */
+export class ScreenReaderAlert extends React.Component<IScreenReaderAlertProps, IScreenReaderAlertState> {
+  // The default props will automatically be applied to this component.
+  public static defaultProps: IScreenReaderAlertProps = defaultScreenReaderAlertProps;
+
+  /**
+   * The index updated during each rendering to make React re-attach the DOM.
+   *
+   * It is critical to make React re-attach the DOM once we want it to read something. Simply updating the
+   * text context inside DOM will cause different behavior in different screen readers or browsers.
+   * Re-attaching the live region element is the most common way which is supported by most major screen reader
+   * and browser combinations.
+   */
+  private _renderIndex: number = 0;
+
+  /**
+   * The non-empty string which represents the latest string readout by screen readers.
+   *
+   * The reason of having this variable is that we have a timeout to clear the text in the live region.
+   * We shouldn't let screen readers read the same string if the indicator is not changed. Use this to let React
+   * render the live region only when text passed from props is changed or the indicator is changed.
+   */
+  private _previousValidString: string = '';
+
+  private _clearTextTimeout: number;
+
+  constructor(props: IScreenReaderAlertProps) {
+    super(props);
+
+    this.state = {
+      alertText: this._getTextFromProps(React.Children.toArray(props.children))
+    };
+  }
+
+  public componentWillReceiveProps(nextProps: IScreenReaderAlertProps): void {
+    if (this._clearTextTimeout) {
+      // Clear the timeout of removing text to avoid the new text being removed.
+      window.clearTimeout(this._clearTextTimeout);
+      this._clearTextTimeout = undefined;
+    }
+
+    this.setState({
+      alertText: this._getTextFromProps(React.Children.toArray(nextProps.children))
+    });
+  }
+
+  public shouldComponentUpdate(nextProps: IScreenReaderAlertProps, nextState: IScreenReaderAlertState): boolean {
+    // When indicator is not changed, do NOT need to render the string if it's same with the previous non-empty string.
+    const isValidStringChanged: boolean = nextState.alertText !== this._previousValidString;
+
+    return (this.props.indicator !== nextProps.indicator || isValidStringChanged);
+  }
+
+  public componentDidUpdate(nextProps: IScreenReaderAlertProps): void {
+    this._clearTextByTimeout();
+    this._updateValidString();
+  }
+
+  public componentDidMount(): void {
+    this._clearTextByTimeout();
+    this._updateValidString();
+  }
+
+  public render(): JSX.Element {
+    return (
+      <div className={ styles.screenReaderAlert } key={ this._renderIndex++ }>
```

Comment:
It has some description above **_renderIndex**. The key is critical to make the text re-attach to the DOM to make sure different screen reader and browser combination can read the text properly.